Source/Core/Core/HW/DSPHLE/UCodes/ASnd.cpp

```diff
@@ -0,0 +1,568 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+// High-level emulation for the libasnd ucode, used by older homebrew
+// libasnd is copyright 2008 Hermes <www.entuwii.net> and released under the BSD-3-Clause license
+
+#include "Core/HW/DSPHLE/UCodes/ASnd.h"
+
+#include <algorithm>
+
+#include "Common/ChunkFile.h"
+#include "Common/Logging/Log.h"
+#include "Common/Swap.h"
+#include "Core/HW/DSP.h"
+#include "Core/HW/DSPHLE/DSPHLE.h"
+#include "Core/HW/DSPHLE/MailHandler.h"
+#include "Core/HW/DSPHLE/UCodes/UCodes.h"
+
+namespace DSP::HLE
+{
+// "fill the internal sample buffer and process the voice internally"
+constexpr u32 MAIL_INPUT_SAMPLES = 0x0111;
+// "get samples from the external buffer to the internal buffer and process the voice mixing the
+// samples internally" (not used)
+constexpr u32 MAIL_INPUT_SAMPLES_2 = 0x0112;
+// "get the address of the voice datas buffer (CHANNEL DATAS)" (actually sets it)
+constexpr u32 MAIL_SET_VOICE_DATA_BUFFER = 0x0123;
+// "process the voice mixing the samples internally"
+constexpr u32 MAIL_INPUT_NEXT_SAMPLES = 0x0222;
+// "send the samples for the internal buffer to the external buffer"
+constexpr u32 MAIN_SEND_SAMPLES = 0x0666;
+// "special: to dump the IROM Datas (remember disable others functions from the interrupt vector to
+// use) (CMBH+0x8000) countain the address of IROM" (not used)
+constexpr u32 MAIL_ROM_DUMP_WORD = 0x0777;
+// "Used for test" (not used)
+constexpr u32 MAIL_TEST = 0x0888;
+constexpr u32 MAIL_TERMINATE = 0x0999;
+
+// Note that there are additional flags used on the powerpc side (UPDATEADD, UPDATE, and VOLUPDATE)
+// that are not relevant to the DSP side.
+// The old flags are used in the 2008 and 2009 versions, while the new ones are used in the 2011 and
+// 2020 versions.
+constexpr u32 OLD_FLAGS_VOICE_PAUSE = 1 << 5;
+constexpr u32 OLD_FLAGS_VOICE_LOOP = 1 << 2;
+constexpr u32 OLD_FLAGS_SAMPLE_FORMAT_MASK = 3;
+
+constexpr u32 NEW_FLAGS_VOICE_PAUSE = 1 << 9;
+constexpr u32 NEW_FLAGS_VOICE_LOOP = 1 << 8;
+constexpr u32 NEW_FLAGS_SAMPLE_FORMAT_MASK = 7;
+// Used on the PowerPC side as an enabled flag (if 0, then the voice isn't active).
+// Used on the DSP side as the step in bytes after each read.
+constexpr u32 FLAGS_SAMPLE_FORMAT_BYTES_MASK = 0xffff0000;
+constexpr u32 FLAGS_SAMPLE_FORMAT_BYTES_SHIFT = 16;
+
+// November 14, 2008 version (padded to 0x05a0 bytes) - initial release
+// https://github.com/devkitPro/libogc/compare/c76d8b851fafc11b0a5debc0b40842929d5a5825~...353a44f038e75e5982eb550173ec8127ab35e3e3
+constexpr u32 HASH_2008 = 0x8d69a19b;
+// February 5, 2009 version (0x05ac bytes) - added MAIL_TERMINATE
+// https://github.com/devkitPro/libogc/compare/1925217ffb4c97cbee5cf21fa3c0231029b340e2~...3b1f018dbe372859a43bff8560e2525f6efa4433
+constexpr u32 HASH_2009 = 0xcc2fd441;
```

Comment:
I never updated this value (or the corresponding size) for https://github.com/dolphin-emu/dolphin/pull/10763/files/ffc2d5c9d5e40a44289e787d419ee0e24fb3d870#diff-e1a93814bdc91cfe049538302fbbd460ff0d679c2560825b1834409a47b16e42. However, `HASH_2009` is never actually read; only `HASH_2008`, `HASH_2011`, and `HASH_2020` are (2009 is the middle release while the first and last releases each have behavior we can conditionally enable), so this doesn't cause problems; this is something that I should fix when adding aesndlib, but it doesn't need to be fixed urgently (unlike the build errors).

Reply:
After checking this again, I _did_ already update the value; it's just size in the comment that's outdated.